stdlib/arg.ml

```diff
@@ -336,3 +336,46 @@ let align ?(limit=max_int) speclist =
   let len = List.fold_left max_arg_len 0 completed in
   let len = min len limit in
   List.map (add_padding len) completed
+
+
+let read_aux sep file =
+  let ic = open_in_bin file in
+  let buf = Buffer.create 200 in
+  let words = ref [] in
+  let stash inw =
+    if inw then begin
+      words := Buffer.contents buf :: !words;
+      Buffer.clear buf;
+    end in
+  let rec read inw =
+    try
+      let c = input_char ic in
+      if c = sep then begin
+        stash inw; read true
```

Comment:
This should be `read false`, right?
In fact you should drop `inw` altogether because we decided we want to allow empty arguments. Just add a test at EOF to stash only when the buffer is not empty.